packages/snowpack/src/build/build-pipeline.ts

```diff
@@ -63,12 +68,40 @@ async function runPipelineLoadStep(
       const mainOutputExt = step.resolve.output[0];
       return {[mainOutputExt]: result};
     } else if (result && typeof result === 'object') {
+      result;
+
+      // handle source maps
+      Object.keys(result).forEach((ext) => {
+        const output = result[ext];
+        if (typeof output !== 'object' || !output.code) return;
+
+        if (output.map) {
+          // some plugins return map as an object; others as a string. handle both cases
+          result[ext + '.map'] =
```

Comment:
coming back after reading through the whole PR: I don't like the idea of managing both `output[ext+'.map']` and `output[ext] = {code, map}`. This feels wrong to me in two ways:

1. It's unclear which is the "right" abstraction at any one time. For example, the same source map appears to live in two places at any one time: its own `output` object entry and as a property on the original `ext` data object.
2. By keeping it in as an entry in the output object we need to treat it as a top-level file. This means some hacks in config.ts to add new `foo.map` entries to `resolve.output` automatically, source maps go through the `transform()` function now, etc.

After reading through this, I think keeping source map as a property on the original entry object is the better path forward. It gives us a single source of truth, fewer (no?) new hacks for how we treat source maps as special in the output object, and makes it much easier to manage the source map (get it directly from the entry, clear it out if a second transform makes it outdated, etc. etc.)

Reply:
Yeah, I don’t disagree. Some part of me doesn’t like the “meta” files such as proxy files and source maps, because it doesn’t neatly fit in our build output structure. I wanted Snowpack to have a “dumb” awareness of these files by simply existing in the output map same as everything else.

However, you are right that by their very nature they are still meta files, and never exist independently apart from their sources. I’m happy to take them back out of the build map. But I think I’m still glad I started this way, for nothing else than for understanding. It does seem like either way you’re still dealing with conditional logic, so we may as well treat special files as special and be more explicit about it.